source/protobuf_restricted/calibrationoperations_restricted.proto

```diff
@@ -0,0 +1,40 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "com.ni.grpc.devicerestricted";
+option java_outer_classname = "NiDeviceRestricted";
+option csharp_namespace = "NationalInstruments.Grpc.DeviceRestricted";
+
+package nidevice_restricted_grpc;
+
+import "deviceid_restricted.proto";
+import "google/protobuf/timestamp.proto";
+
+service CalibrationOperationsRestricted {
+  rpc GetCalibrationInformation(GetCalibrationInformationRequest) returns (GetCalibrationInformationResponse);
+}
+
+message GetCalibrationInformationRequest {
+  DeviceId device_id = 1;
+}
+
+message GetCalibrationInformationResponse {
+  bool supports_internal_calibration_available = 1;
```

Comment:
For these, you can consider using the `optional` field rule for the singular fields instead of an extra `_available` bool for each. I think it would give similar behavior but am not sure for this restricted service situation if it'd be worth it or not.

Reply:
Using `optional` instead of `_available` fields is a great idea.  I completely forgot that this capability was built into protocol buffers.

Reply:
Shoutout to Brad who pointed it out as an option in a recent Scope change. Might be worth noting that it was [added to proto3 with protobuf version v3.15.0 (Early 2021)](https://github.com/protocolbuffers/protobuf/releases/tag/v3.15.0) but we added it in niscope.proto and in general all the protoc's we use should be past that version.

Reply:
Added `optional` to all scalar basic type fields.  The repeated and message fields don't need it because it is already possible to determine their field presence.

I built InstrumentStudio with these changes and the version of the gRPC NuGets we are using has this support.